### PR TITLE
Fix incorrect width/height indexing in v8OBBLoss

### PR DESCRIPTION
**Pull request by raimbekovm**
Fixed swapped dimension indexing when filtering rotated bounding boxes in v8OBBLoss.__call__().

Changes:
- Line 705: Corrected rw, rh calculation to use proper indices from imgsz (h, w) format
- Before: width × height, height × width
- After: width × width, height × height

Aligns with line 707 usage of imgsz[[1, 0, 1, 0]] and v8PoseLoss keypoint scaling pattern.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes swapped image width/height scaling when filtering tiny rotated boxes in `v8OBBLoss` 🧭

### 📊 Key Changes
- Corrected `imgsz` indexing when computing rotated box width/height (`rw`, `rh`) in `ultralytics/utils/loss.py`
- Ensures rbox size filtering uses the proper image dimension scaling before preprocessing

### 🎯 Purpose & Impact
- Prevents incorrect filtering of valid (or invalid) tiny OBB targets due to width/height being scaled against the wrong image side ✅
- Improves training stability and label handling consistency for OBB models, especially on non-square inputs 📦

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message—an Ultralytics engineer will review and assist soon. To ensure a seamless integration of your work, please review the following checklist:

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/utils/loss.py [hunk 1]
```diff
@@ -702,7 +702,7 @@ def __call__(self, preds: Any, batch: dict[str, torch.Tensor]) -> tuple[torch.Te
         try:
             batch_idx = batch["batch_idx"].view(-1, 1)
             targets = torch.cat((batch_idx, batch["cls"].view(-1, 1), batch["bboxes"].view(-1, 5)), 1)
-            rw, rh = targets[:, 4] * imgsz[0].item(), targets[:, 5] * imgsz[1].item()
+            rw, rh = targets[:, 4] * imgsz[1].item(), targets[:, 5] * imgsz[0].item()
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/loss.py]: 📝 **LOW**: This fix correctly swaps `imgsz` indexing for `rw`/`rh` and aligns with the later `scale_tensor=imgsz[[1, 0, 1, 0]]`. One small robustness improvement: if `imgsz` is already a Python tuple/list of ints in some call paths, calling `.item()` will raise. Consider casting via `float(...)` (works for Python numbers and 0-d tensors) or guarding for tensor vs non-tensor.

**Suggested change:**
```suggestion
            rw, rh = targets[:, 4] * float(imgsz[1]), targets[:, 5] * float(imgsz[0])
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, targeted bug fix: swapping `imgsz` indices for rotated box width/height matches the subsequent scaling tensor and should prevent incorrect tiny-rbox filtering on non-square inputs. Only potential issue is `.item()` robustness if `imgsz` can be non-tensor; consider using `float(imgsz[i])` to avoid type-dependent failures.

💬 Posted 1 inline comment

**Review by Laughing-q — APPROVED** [review]

**Review by Laughing-q — APPROVED** [review]
